### ISSUE TITLE
Option to restore/import backup .

### ISSUE THREAD

**Issue by GoogleCodeExporter** [QuiteRSS/quiterss#625]
```
There is a option to create backup but no option to restore/import backup files 
.
```

Original issue reported on code.google.com by `nikhilch...@gmail.com` on 25 Aug 2014 at 7:44

**Comment by logos88** [comment]
I would like to see this too. It´s useless to have a backup option if you don´t have the option to restore it. In windows I can´t do nothing with the archives created with the backup.

**Comment by humaidq** [comment]
I agree, I also need this because I created a backup and I lost all my feeds. But I am not sure if I can get the feed urls from the bin file.

**Comment by DL-webmaster** [comment]
Meanwhile, 5 years later... nada... going to look of for a reader that does both... Just lost 100+ feeds for our news blog and no way to import them. Ciao!

**Comment by humaidq** [comment]
So looking back at this issue. It seems when you create a backup, it simply copies your `QuiteRSS.ini` and `feeds.db`, appending the date and `.bak` extension.  

So simply copying over your:
- `feeds.db.<date>.bak` -> `~/.local/share/QuiteRss/QuiteRss/feeds.db`
- `QuiteRss.ini.<date>.bak` -> `~/.config/QuiteRss/QuiteRss.ini`

I am not sure however the location of these files on Windows, but you might find a "QuiteRss" folder inside `%APPDATA%` or %LOCAL%`.

---

Another way to get the data from the feed backup file (if you no longer have QuiteRSS installed) is to use `sqlite3` and query: `SELECT title,xmlUrl from feeds;`

**Comment by kgkoutis** [comment]
Well.. following what @humaidq mentioned above, if you make a backup of the feeds.db living inside C:\Users\username\AppData\Local\QuiteRss\QuiteRss folder, uninstall everything, reinstall everything, start the program, close it immediately, and restore the backup by overwriting the new copy of feeds.db, that's enough. I just tried it and it found all my feeds as they were. Everything seem to rely just on the db, because I could not find the QuiteRss.ini anywhere.

I wrote the following backup.bat file to zip it, provided it that sqlite3 and 7z are in the PATH

`sqlite3 feeds.db ".backup feeds.db.bak" && 7z a -t7z -m0=lzma -mx=9 -mfb=64 -md=32m -ms=on feeds.db.bak.7z .\feeds.db.bak`

**Comment by Weresmilodon** [comment]
I just had to go though this mess myself. You can find the .ini in \AppData\Roaming\QuiteRss and the .db in \AppData\Local\QuiteRss\QuiteRss. Why they're not in the same folder i don't know.

It's fucking ridiculous that there's no information on how to restore a backup, nevermind a restore function in the program itself.

**Comment by QAQDE** [comment]
For mac user who want to restore the backup, please first of all rename the file to feeds.db instead of feeds.db.date.bk

and put the file into, username/library/application support/quiterss, in this page you would see the feeds.db, and then replace your backup. Finished! Happy life..

**Comment by SavchukR** [comment]
For Windows do the following

1. Close app
2. Db file `feeds.db_[date].bak` - put it to `C:\Users\[username]\AppData\Local\QuiteRss\QuiteRss` folder and renamte to `feeds.db`.
3. Ini file `QuiteRss.ini_[date].bak` - put it to `C:\Program Files (x86)\QuiteRSS` and rename to `QuiteRss.ini`
4. Open application - feed should be there
